goth/runner/process.py

```diff
@@ -13,13 +13,30 @@
 RUN_COMMAND_DEFAULT_TIMEOUT = 900  # seconds
 
 
+class ProcessMonitor:
+    """Monitor enabling acquisition of the process object of a running command."""
+
+    _process: Optional[asyncio.subprocess.Process] = None
+
+    async def get_process(self) -> asyncio.subprocess.Process:
+        """Wait for and return the `Process` object."""
+        while not self._process:
+            await asyncio.sleep(0.1)
+        return self._process
+
+    def set_process(self, process: asyncio.subprocess.Process):
+        """Set the process object."""
+        self._process = process
```

Comment:
In my opinion instances of this class should be considered immutable. This "public" method implies it's valid to set the process from the calling code. I'd suggest removing this method and referencing `_process` directly in `run_command`.

Reply:
okay, I guess